src/txmempool.h

```diff
@@ -623,7 +629,10 @@ class CTxMemPool
     /** Affect CreateNewBlock prioritisation of transactions */
     void PrioritiseTransaction(const uint256& hash, const CAmount& nFeeDelta);
     void ApplyDelta(const uint256& hash, CAmount &nFeeDelta) const EXCLUSIVE_LOCKS_REQUIRED(cs);
-    void ClearPrioritisation(const uint256& hash) EXCLUSIVE_LOCKS_REQUIRED(cs);
+    void ClearPrioritisation(const uint256& hash);
+
+    /** Recompute valuation of all transaction fees, called whenever exchange rates have been updated. */
+    void RecomputeFees();
```

Comment:
Do we need that EXCLUSIVE_LOCKS_REQUIRED here? Why or why not?

Reply:
`RecomputeFees()` _takes_ the lock, and `EXCLUSIVE_LOCKS_REQUIRED` means that the lock must already be held. So we don't need it here.